lib/ansible/modules/cloud/amazon/iam_managed_policy.py

```diff
@@ -102,137 +102,187 @@
   returned: success
   type: string
   sample: '{
-        "Arn": "arn:aws:iam::aws:policy/AdministratorAccess "
-        "AttachmentCount": 0,
-        "CreateDate": "2017-03-01T15:42:55.981000+00:00",
-        "DefaultVersionId": "v1",
-        "IsAttachable": true,
-        "Path": "/",
-        "PolicyId": "ANPALM4KLDMTFXGOOJIHL",
-        "PolicyName": "AdministratorAccess",
-        "UpdateDate": "2017-03-01T15:42:55.981000+00:00"
+        "arn": "arn:aws:iam::aws:policy/AdministratorAccess "
+        "attachment_count": 0,
+        "create_date": "2017-03-01T15:42:55.981000+00:00",
+        "default_version_id": "v1",
+        "is_attachable": true,
+        "path": "/",
+        "policy_id": "ANPALM4KLDMTFXGOOJIHL",
+        "policy_name": "AdministratorAccess",
+        "update_date": "2017-03-01T15:42:55.981000+00:00"
   }'
 '''
+
 from ansible.module_utils.basic import AnsibleModule
-import ansible.module_utils.ec2
-from ansible.module_utils.ec2 import sort_json_policy_dict
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import sort_json_policy_dict, camel_dict_to_snake_dict, HAS_BOTO3
 import json
+import traceback
 
 try:
-    import boto3
     import botocore
-    HAS_BOTO3 = True
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by imported HAS_BOTO3
 
 
-def get_policy_by_name(iam, name, **kwargs):
-    response = iam.list_policies(Scope='Local', **kwargs)
+def get_policy_by_name(module, iam, name, **kwargs):
+    try:
+        response = iam.list_policies(Scope='Local', **kwargs)
```

Comment:
Good point. Easiest to fix using the paginator with build_full_result, along with AWSRetry